pkg/cmd/repo/deploy-key/add/add.go

```diff
@@ -0,0 +1,112 @@
+package add
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/cli/cli/v2/internal/config"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type AddOptions struct {
+	IO         *iostreams.IOStreams
+	Config     func() (config.Config, error)
+	HTTPClient func() (*http.Client, error)
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	KeyFile   string
+	Title     string
+	ReadWrite bool
+}
+
+func NewCmdAdd(f *cmdutil.Factory, runF func(*AddOptions) error) *cobra.Command {
+	opts := &AddOptions{
+		HTTPClient: f.HttpClient,
+		Config:     f.Config,
+		IO:         f.IOStreams,
+		BaseRepo:   f.BaseRepo,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "add [<key-file>] [--read-write]",
+		Short: "Add a deploy key to your GitHub repository",
```

Comment:
Since the command operates on the repository of the current repository by default (unless `--repo` was specified), let's consistently say "to *the* repository" instead of "to _your_ GitHub repository".